internal/exec/stages/disks/luks.go

```diff
@@ -306,24 +330,52 @@ func (s *stage) createLuks(config types.Config) error {
 				}
 			}
 
+			// additionally if there are tang servers with advertisement, lets see if they are reachable, and check to make sure the keys match
+			for _, tang := range tangServersWithAdvertisement {
+				u, err := url.Parse(tang.URL)
+				if err != nil {
+					return fmt.Errorf("parsing tang URL: %v", err)
+				}
+				u.Path = path.Join(u.Path, "adv")
+				data, err := s.Fetcher.FetchToBuffer(*u, resource.FetchOptions{})
+
+				if err != nil {
+					// this is somewhat expected, server is likely not reachable
+					continue
+				}
+
+				fetchedAdv := string(data)
+				// if the server is reachable, lets check to make sure the keys match
+				if fetchedAdv != *tang.Advertisement {
+					return fmt.Errorf("tang advertisement does not match: %v", err)
+				}
+			}
+
+			// lets bind our device
 			if _, err := s.Logger.LogCmd(
 				exec.Command(distro.ClevisCmd(), "luks", "bind", "-f", "-k", keyFilePath, "-d", devAlias, pin, config), "Clevis bind",
 			); err != nil {
 				return fmt.Errorf("binding clevis device: %v", err)
 			}
 
-			// close & re-open Clevis devices to make sure that we can unlock them
-			if _, err := s.Logger.LogCmd(
-				exec.Command(distro.CryptsetupCmd(), "luksClose", luks.Name),
-				"closing clevis luks device %v", luks.Name,
-			); err != nil {
-				return fmt.Errorf("closing luks device: %v", err)
-			}
-			if _, err := s.Logger.LogCmd(
-				exec.Command(distro.ClevisCmd(), "luks", "unlock", "-d", devAlias, "-n", luks.Name),
-				"reopening clevis luks device %s", luks.Name,
-			); err != nil {
-				return fmt.Errorf("reopening luks device %s: %v", luks.Name, err)
+			// since the number of tang servers that are "online" is greater then the threshold, we can safely close and re-open the device
+			// otherwise we need to do an offline first boot and forgo the close and re-open
+			if len(tangServersWithoutAdvertisement) > *luks.Clevis.Threshold {
```

Comment:
We also need to account for the TPM here.

Are you planning to pursue the additional step proposed in https://github.com/coreos/ignition/issues/1474#issuecomment-1424745135 and do the reopen if enough `tangServersWithAdvertisement` are up?